src/main/java/com/redhat/cloud/notifications/db/EndpointResources.java

```diff
@@ -78,8 +83,8 @@
         }
 
         return mutinyQuery.getResultList()
-                .onItem().transformToMulti(Multi.createFrom()::iterable)
-                .onItem().transformToUniAndConcatenate(this::loadProperties);
+                .onItem().call(this::loadProperties)
+                .onItem().transformToMulti(Multi.createFrom()::iterable);
```

Comment:
The `Uni` -> `Multi` transformation can probably be avoided in most cases thanks to your idea.